### ISSUE TITLE
AuthorizationException: {"type":4,"code":4001,"error":"invalid_redirect_uri"}

### ISSUE THREAD

**Issue by ryanchyshyn** [okta/okta-oidc-android#141]
Calling several WebAuthClient methods will cause:

> AuthorizationException: {"type":4,"code":4001,"error":"invalid_redirect_uri"}
>         at com.okta.oidc.util.AuthorizationException$RegistrationRequestErrors.<clinit>(AuthorizationException.java:433)
>         at com.okta.oidc.clients.web.SyncWebAuthClientImpl.processSignOutResult(SyncWebAuthClientImpl.java:369)
>         at com.okta.oidc.clients.web.SyncWebAuthClientImpl.signOutOfOkta(SyncWebAuthClientImpl.java:347)
>         at com.okta.oidc.clients.web.WebAuthClientImpl.lambda$signOutOfOkta$8$WebAuthClientImpl(WebAuthClientImpl.java:188)
>         at com.okta.oidc.clients.web.-$$Lambda$WebAuthClientImpl$IIx9nwOBUzFH6kYpa3Zs5czCKyQ.run(lambda)
>         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:423)
>         at java.util.concurrent.FutureTask.run(FutureTask.java:237)
>         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
>         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
>         at java.lang.Thread.run(Thread.java:818)

The problem is that future that was used previously gets cancelled even if it is completed:
```
class WebAuthClientImpl {

    private void cancelFuture() {
        if (mFutureTask != null && (!mFutureTask.isDone() || !mFutureTask.isCancelled())) {
            mFutureTask.cancel(true);
        }
    }
```
In my case isDone = true, isCancelled = false. Possible fix - replace || with &&.

**Comment by FeiChen-okta** [comment]
Hi @ryanchyshyn Thanks for the analysis. We will take a look at this issue.

**Comment by congybk** [comment]
@FeiChen-okta I also had the same problem
{"type":4,"code":4001,"error":"invalid_redirect_uri","errorDescription":"No uri registered to handle redirect or multiple applications registered"}

**Comment by PauloLinhares09** [comment]
I have the same problem, and it not fix for me ! Somebody can help ?

**Comment by FeiChen-okta** [comment]
Fixed with 1.0.14 release.

**Comment by coreform** [comment]
I see this problem on Samsung Galaxy S6 and Samsung Galaxy Note5 devices, it seems that the new okta-oidc-android library has not got the provisions for the known problems on those devices that were added to the old okta-appauth-android library. https://github.com/okta/okta-oidc-android/issues?q=is%3Aissue+https%3A%2F%2Fgithub.com%2Fopenid%2FAppAuth-Android%2Fissues%2F157
